stat/stat.go

```diff
@@ -135,6 +139,55 @@ func (s *StatterWithTimeout) doLstat(path string, infoCh chan fs.FileInfo, errCh
 	}
 
 	info, err := s.lstat(path)
+	if err == nil {
+		stat, ok := info.Sys().(*syscall.Stat_t)
+		if ok {
+			correctZeroTimes(stat, s.defTime, path, s.lstat)
+
+			correctFutureTimes(stat, s.defTime)
+		}
+	}
+
 	infoCh <- info
 	errCh <- err
 }
+
+func correctZeroTimes(stat *syscall.Stat_t, now int64, path string, lstat LstatFunc) {
```

Comment:
This should be a method on StatterWithTimeout so you don't need to pass now and Lstat